lib/tabular_form_builder.rb

```diff
@@ -198,10 +198,29 @@ def label_for_field(field, options = {}, translation_form = nil)
 
     id = element_id(translation_form) if translation_form
 
-    # FIXME: reenable the error handling
-    label_options[:class] << 'error' if false && @object && @object.respond_to?(:errors) && @object.errors[field] # FIXME
     label_options[:class] << 'form--label'
-    label_options[:class] << ' -required' if options.delete(:required)
+
+
+    content = h(text)
+    if @object.try(:errors) && @object.errors.include?(field)
+      label_options[:class] << ' -error'
+      error_label = I18n.t('errors.field_erroneous_label',
+                           full_errors: @object.errors.full_messages_for(field).join(' '))
+       content << content_tag('p', error_label, class: 'hidden-for-sighted')
+    end
+
+    if options.delete(:required)
+      label_options[:class] << ' -required'
+      content << content_tag('span',
+                             '*',
+                             class: 'form--label-required',
+                             'aria-hidden': true)
+
+      content << content_tag('p',
+                             I18n.t(:label_field_is_required),
+                             class: 'hidden-for-sighted')
```

Comment:
Is this necessary if the input that is labeled has the html5 "required" attribute?